xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -400,17 +440,26 @@ PVR_ERROR CPVRClients::GetCreatedClients(CPVRClientMap& clientsReady, std::vecto
 
   for (const auto& addon : addons)
   {
-    int iClientId = ClientIdFromAddonId(addon->ID());
-    std::shared_ptr<CPVRClient> client;
-    GetClient(iClientId, client);
-
-    if (client && client->ReadyToUse() && !client->IgnoreClient())
-    {
-      clientsReady.insert(std::make_pair(iClientId, client));
-    }
+    std::vector<uint32_t> instanceIds;
+    if (addon->SupportsMultipleInstances())
+      instanceIds = addon->GetKnownInstanceIds();
     else
+      instanceIds.emplace_back(ADDON_SINGLETON_INSTANCE_ID);
```

Comment:
I see this code sequence twice. Would it make sense to extend `CAddon::GetKnownInstanceIds`to contain the whole sequence duplicated here?